yellowbrick/cluster/elbow.py

```diff
@@ -262,6 +270,9 @@ def fit(self, X, y=None, **kwargs):
                 self.scoring_metric(X, self.estimator.labels_)
             )
 
+        self.kneedle=KneeLocator(self.k_values_,self.k_scores_,curve='convex',direction='decreasing')
+        self.knee_value=self.kneedle.find_knee()[0]
```

Comment:
After taking a look at the `KneeLocator` code, I noticed that `self.knee, self.norm_knee, self.knee_x` are all set when the locator is initialized at `__init__`() therefore there is no need to call `find_knee()` again (which is potentially an expensive operation). You can just do `self.elbow_value_ = knee_locator.knee`. 

Also below you use `self.k_values_.index`, you may be able to use `locator.knee_x` instead, but I'm not 100% sure about that.

Reply:
@bbengfort Thanks for such in-depth review pointing out these small mistakes. Greatly appreciated. Yes you are right we can use `knee_locator.knee_x` instead of accessing the index of the knee value.